python/pyarrow/tests/test_io.py

```diff
@@ -424,6 +424,16 @@ def ne(a, b):
     eq(buf2, buf5)
 
 
+def test_buffer_eq_bytes():
+    buf = pa.py_buffer(b'some data')
+    assert buf == b'some data'
+    assert buf == bytearray(b'some data')
+    assert buf != b'some dat1'
+
+    with pytest.raises(TypeError):
+        buf == 'some data'
```

Comment:
I think this won't raise on py2. You need to use an explicit unicode string (`u'some data'`).

Reply:
Thanks. Counting down the days